HomeWork/Home6.1/MessageDispatcher.java

```diff
@@ -0,0 +1,60 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class MessageDispatcher {
+    private Map<String, List<Subscriber<?>>> subscribers;
+
+    public MessageDispatcher() {
+        this.subscribers = new HashMap<>();
+    }
+
+    public UUID subscribeToGreen(Subscriber<GreenMessage> subscriber) {
+        return subscribe(GreenMessage.class.getSimpleName(), subscriber);
+    }
+
+    public UUID subscribeToBlue(Subscriber<BlueMessage> subscriber) {
+        return subscribe(BlueMessage.class.getSimpleName(), subscriber);
+    }
+
+    public UUID subscribeToOrange(Subscriber<OrangeMessage> subscriber) {
+        return subscribe(OrangeMessage.class.getSimpleName(), subscriber);
+    }
+
+    private UUID subscribe(String messageType, Subscriber<?> subscriber) {
+        UUID subscriptionId = UUID.randomUUID();
+        subscribers.computeIfAbsent(messageType, k -> new ArrayList<>()).add(subscriber);
+        return subscriptionId;
+    }
+
+    public void unsubscribe(UUID subscriptionId) {
+        subscribers.values().forEach(subscriberList -> {
+            subscriberList.removeIf(subscriber -> subscriber.getId().equals(subscriptionId));
+        });
+    }
+
+    public void publish(GreenMessage message) {
+        notifySubscribers(GreenMessage.class.getSimpleName(), message);
```

Comment:
the same concern as for c++ version - we should post it to the queue and handle it with a separate thread